lib/ukio/apartments/handlers/availability_checker.ex

```diff
@@ -0,0 +1,20 @@
+defmodule Ukio.Apartments.Handlers.AvailabilityChecker do
+  import Ecto.Query, warn: false
+
+  alias Ukio.Apartments.Booking
+  alias Ukio.Repo
+
+  def is_apartment_available?(apartment_id, check_in, check_out) do
```

Comment:
We check if:
- there are bookings that will overlap
- if we don't have any (`Enum.empty?()`) we know the apartment is available